### PR TITLE
[AAP-9301] Added msg/var support in debug action

### PR DESCRIPTION
**Pull request by mkanoor**
https://issues.redhat.com/browse/AAP-9301

The debug action supports msg and var attributes to display single/multiple messages or variables akin to ansible debug builtin. The msg supports an array of messages.
This removes the need for a separate echo command. The echo is equivalent to
```
action:
   debug:
     msg: .....
```

### CODE REVIEW

File: docs/actions.rst [hunk 1]
```diff
@@ -360,31 +359,56 @@ When a rule's condition(s) are satisfied we get the results back as:
...
+               - "Message 1 {{ event }}"
+               - Second Message
+
+    .. code-block:: yaml
+
+       name: debug with var
+       condition: event.i >= 5
+       action:
+          debug:
+            var: "{{ event.i }}"
```
Comment (bzwei) [inline docs/actions.rst:408]: From this example and line 399 I don't see the difference var vs msg. Why not just use msg?

Comment (mkanoor) [reply docs/actions.rst:408]: @bzwei var doesn't need the curly braces, we should be able to just use event.i. And use dpath to pick up the data.
In the ansible documentation it says that var should not be using curly braces.
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/debug_module.html#parameter-var
I am going to be replacing the braces and just running dpath on anything in that field. By the time it comes to the debug method it has already gone thru jinja substitution.

Comment (mkanoor) [reply docs/actions.rst]: @bzwei I have removed the {{ }} which is optional

**Review by bzwei — APPROVED** [review]

File: docs/actions.rst [hunk 2]
```diff
@@ -360,31 +359,57 @@ When a rule's condition(s) are satisfied we get the results back as:
...
      - Description
      - Required
-   * - message
-     - A terse message to display to stdout
-     - Yes
+   * - msg
+     - A simple string or an array of strings, which can have references to event or events
+     - No
+   * - var
+     - The variable to print, which can have references to event or events. Using {{ }} is optional.
```
Comment (Alex-Izquierdo) [inline docs/actions.rst:373]: As far as I can see in the code, curly brackets are not supported here.

File: ansible_rulebook/schema/ruleset_schema.json [hunk 3]
```diff
@@ -125,8 +124,7 @@
                         {"$ref": "#/$defs/print-event-action"},
                         {"$ref": "#/$defs/debug-action"},
                         {"$ref": "#/$defs/none-action"},
-                        {"$ref": "#/$defs/shutdown-action"},
-                        {"$ref": "#/$defs/echo-action"}
```
Comment (Alex-Izquierdo) [inline ansible_rulebook/schema/ruleset_schema.json:129]: We should not remove the echo action here if the we don't remove it from the rest of the code. I suggest leaving it for a while, and given the case, remove it completely if a later PR.

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]
Changelog must be updated as well.

File: docs/actions.rst (hunk 2 above)
Comment (mkanoor) [reply docs/actions.rst:373]: @Alex-Izquierdo Every action args goes thru a substitution first and anything that is left is handled in the var for a debug.

**Comment by mkanoor** [comment]
@Alex-Izquierdo Changeling updated

**Review by Alex-Izquierdo — DISMISSED** [review]

**Comment by mkanoor** [comment]
@Alex-Izquierdo @ttuffin Please review I had to rebase to align changes in CHANGELOG.md

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by ttuffin — APPROVED** [review]

**Pull request merged**